### PR TITLE
ZK-Snarks tutorial

### PR DESCRIPTION
**Pull request by bufta**
ZK-Snarks tutorial grant proposal

#### Project Abstract

Main goal of this project is to popularise the ZK-Snarks concept within the Substrate community. We would like to create a series of blog posts with the tutorial codebase, which will help in better understanding the ZK-Snarks for the Substrate community.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md).
- [x] A BTC, Ethereum (USDT/USDC/DAI) or Polkadot/Kusama (aUSD) address for the payment of the milestones is provided inside the application.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. We usually don’t fund purely educational-focused milestones. Could you potentially combine milestones 2 and 3? Also could you the default deliverables to all milestones (0a. License, etc.)? And just to double-check, you want to essentially update https://github.com/zeropoolnetwork/zeropool-substrate-groth16-example, correct?

**Comment by bufta** [comment]
Hi @Noc2  thank you for the feedback. Indeed, coding wise we would like to implement, from scratch, an updated version of the zeropool project. This experience will allow us to document and share the step-by-step guidance for Substrate community (blogposts & video). As for the milestones, we will rearrange the blog posts/video plans to combine the educational efforts with the development.

**Review by Noc2 — DISMISSED** [review]
Thanks for the update. I’m happy to go ahead with it and share it with the rest of the team.

**Comment by semuelle** [comment]
Thanks for the application, @bufta. Just to make sure I understand correctly: you will build and publish a fully functioning zk-snarks pallet and demo (on, say, Github) and then accompanying material on your blog and YouTube?

**Comment by bufta** [comment]
@semuelle as we described it in the proposal, the pallet will be used for the verification process. Based on the implementation part we'll prepare the tutorials (blog posts and video).

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks for the update. Could you also remove the boilerplate text from the `0` deliverables and replace it with more concrete specifications?

**Comment by bufta** [comment]
@semuelle the requested change was applied in the latest commit, hope the 0 deliverables are more suitable now

**Review by semuelle — DISMISSED** [review]
Thanks for the update. I approve.

**Review by Noc2 — DISMISSED** [review]

File: applications/ZK-Snarks tutorial.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# ZK-Snarks tutorial
+* **Proposer:** Bright Inventions
```
Comment (takahser) [inline applications/ZK-Snarks tutorial.md:2]:
```suggestion
* **Team Name:** Bright Inventions
```

File: applications/ZK-Snarks tutorial.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+* **Projects you think this work could be useful for:** Community
+* **Payment Address:** 0xDDcfd2a1D4Bcf058cf2828eb78dB59b300e80F8d
```
Comment (takahser) [inline applications/ZK-Snarks tutorial.md:4]:
```suggestion
* **Payment Address:** 0xDDcfd2a1D4Bcf058cf2828eb78dB59b300e80F8d
- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 2

```

File: applications/ZK-Snarks tutorial.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
+# ZK-Snarks tutorial
+* **Proposer:** Bright Inventions
+* **Projects you think this work could be useful for:** Community
```
Comment (takahser) [inline applications/ZK-Snarks tutorial.md:3]:
```suggestion
```

**Review by takahser — CHANGES_REQUESTED** [review]
@bufta please make sure that your application adheres to our [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md).

**Comment by bufta** [comment]
Thank you @takahser, I applied the suggested changes

**Review by Noc2 — APPROVED** [review]

**Review by semuelle — APPROVED** [review]

**Comment by takahser** [comment]
@bufta thanks for the update.
@alxs could you have a look at this application, since it focuses on zero-knowledge-based tech?

**Review by laboon — APPROVED** [review]

**Pull request merged**